packages/server/database/types/processTeamsLimitsJob.ts

```diff
@@ -0,0 +1,33 @@
+import {r} from 'rethinkdb-ts'
+import sendTeamsLimitEmail from '../../billing/helpers/sendTeamsLimitEmail'
+import {DataLoaderWorker} from '../../graphql/graphql'
+import isValid from '../../graphql/isValid'
+import ScheduledTeamLimitsJob from './ScheduledTeamLimitsJob'
+
+const processTeamsLimitsJob = async (job: ScheduledTeamLimitsJob, dataLoader: DataLoaderWorker) => {
+  const {orgId, runAt, type} = job
+  const [organization, orgUsers] = await Promise.all([
+    dataLoader.get('organizations').load(orgId),
+    dataLoader.get('organizationUsersByOrgId').load(orgId)
+  ])
+  const {name: orgName, scheduledLockAt, lockedAt} = organization
+
+  // Skip the job if it's already been locked or the scheduled lock date changed
+  if (!scheduledLockAt || lockedAt || scheduledLockAt.getTime() !== runAt.getTime()) return
```

Comment:
`scheduledLockAt.getTime() !== runAt.getTime()` 
This check wouldn't work as we have `runAt = scheduledLockAt.getTime() - ms('7d')`
Maybe we need to store `scheduledFirstWarningAt` value

Reply:
Makes sense. I've updated this so that we have an `expectedRunAt` timestamp based on the job type